### ISSUE TITLE
ArucoInvoke.DrawCharucoDiamond results in CvException

### ISSUE THREAD

**Issue by jbruening** [emgucv/emgucv#173]
Exception is:  
```Emgu.CV.Util.CvException: 'OpenCV: d == 2 && ((sizes[0] == sz.height && sizes[1] == sz.width) || (allowTransposed && sizes[0] == sz.width && sizes[1] == sz.height))'```

Knowing that DrawCharucoDiamond just generates a CharucoBoard of 3x3 and modifies the board.ids vector to be the ids parameter of DrawCharucoDiamond, and then calls board.Draw, this doesn't really make sense as to why it's failing, as board.Draw works fine (albeit with no modified ids). [C++ I'm referring to](https://github.com/opencv/opencv_contrib/blob/master/modules/aruco/src/charuco.cpp#L868)

test code:
``` csharp  
var SquareSize = 200;
var MarkerSize = 120;

using (var dict = new Dictionary(Dictionary.PredefinedDictionaryName.Dict4X4_250))
using (var ids = new VectorOfInt(new[] { 0, 1, 2, 3 }))
using (var board = new CharucoBoard(3, 3, SquareSize, MarkerSize, dict))
{
    using (var img = new Mat())
    {
        //c# equivilent of DrawCharucoDiamond, but board.ids is not exposed
        //for (var i = 0; i < 4; i++)
        //board.ids[i] = ids[i];

        var size = 3 * SquareSize + 2 * MarkerSize;
        var outSize = new Size(size, size);
        //works just fine
        board.Draw(outSize, img, 0, 1);
    }

    using (var img = new Mat())
    {
        //but this throws an exception...
        ArucoInvoke.DrawCharucoDiamond(dict, ids, SquareSize, MarkerSize, img, 0, 1);
    }
}
```

**Comment by emgucv** [comment]
Fixed in the above commit. 

Thanks for pointing out. 

Tested to work with the following code:

```
            String win1 = "Test Window"; //The name of the window
            CvInvoke.NamedWindow(win1); //Create the window using the specific name
            var SquareSize = 200;
            var MarkerSize = 120;
            using (var dict = new Dictionary(Dictionary.PredefinedDictionaryName.Dict4X4_250))
            using (var board = new CharucoBoard(3, 3, SquareSize, MarkerSize, dict))
            {
                using (var img = new Mat())
                {
                    var size = 3 * SquareSize + 2 * MarkerSize;
                    var outSize = new Size(size, size);
                    //works just fine
                    board.Draw(outSize, img, 0, 1);
                }
                using (var img = new Mat())
                {
                    ArucoInvoke.DrawCharucoDiamond(dict, new[] { 0, 1, 2, 3 }, SquareSize, MarkerSize, img, 0, 1);
                    CvInvoke.Imshow(win1, img); //Show the image
                    CvInvoke.WaitKey(0);  //Wait for the key pressing event
                    CvInvoke.DestroyWindow(win1); //Destroy the window if key is pressed
                }
            }
```

Will be included in the 4.0 release.

**Comment by emgucv** [comment]
Closing ticket now.
